### PR TITLE
Document and shorten the two vmex-ESSOS Python seams

### PR DESCRIPTION
**Pull request by rogeriojorge**
## The seams

vmex and ESSOS meet at exactly two Python seams, each one-directional. Before
this PR a user had to work out which object to build, which class consumed it,
and which way the data flowed, from example sources.

| Direction | Before | After |
| --- | --- | --- |
| vmex equilibrium -> ESSOS | `write_wout(tmp)` then `essos.fields.Vmec(str(tmp))`, keeping the temporary alive by hand | `vj.essos_vmec_field(wout_or_path)` |
| ESSOS coils -> vmex | `jax.jit(jax.vmap(BiotSavart(coils).B))`, a chunking wrapper, hand-picked grid bounds, then `MgridField.from_cartesian_field` | `vj.MgridField.from_coils(coils)` |

There is no third seam. `MgridField` -> coils is a coil-design inverse problem
and stays in ESSOS; that is said in the docs rather than implemented.

## What changed

- **`vmex/core/tracing.py`** — new `essos_vmec_field(source, **kwargs)`. It does
  the temporary-wout hop released ESSOS needs (ESSOS reads every table eagerly
  in its constructor, so the file is gone by the time the field is returned) and
  carries the lasym rejection. `trace_alphas` now calls it instead of repeating
  the hop, so the module has one handoff, not two. No new module: this file was
  already the ESSOS adapter.
- **`vmex/core/mgrid.py`** — new `MgridField.from_coils(coils, ...)`, next to
  `from_cartesian_field`. Bounds default to the coil bounding box grown by 10%,
  `nfp` to the coil set's own. Accepts a `Coils` or a `BiotSavart`.
- **`vmex/core/mgrid.py`** — the tabulator's ESSOS branch. `BiotSavart.B` takes
  one point, and `_cartesian_field_values` fell back to a Python loop over every
  grid point: **294912 points took ~200 s, against ~0.2 s** vmapped in chunks
  sized from the coil array. `vmex --coils` took that slow path. It now batches.
- **`vmex/core/cli.py`** — `_coils_mgrid_field` is file loading plus
  `MgridField.from_coils`; 25 lines of duplicated bounding-box and Biot-Savart
  wiring deleted, and the CLI inherits the batching fix.
- **`examples/free_boundary_essos_coils.py`** — drops its hand-rolled
  vmap-and-chunk wrapper for `from_coils` with the same explicit bounds. Same
  grid, same numbers.
- **`examples/vmex_essos_workflow.py`** (new) — walks both seams and closes the
  loop: solve a fixed boundary, hand it to ESSOS, measure iota from an ESSOS
  field-line trace, tabulate the ESSOS coil set, solve a vacuum free boundary
  with it, hand that equilibrium back across the same seam.
- **`docs/howto/use-essos-fields-and-coils.md`** (new) — one page, both
  directions stated plainly, with the measured accuracy of each and the three
  caveats that bite (the write severs the gradient; stellarator symmetry only;
  ESSOS interpolates the half-mesh tables linearly in `s`, so the two `|B|`
  channels differ by 5.9% at `ns = 16` and 1.7% at `ns = 51`).

Only the released ESSOS surface is used throughout. Nothing here depends on
uwplasma/ESSOS#61.

## What the example measures

```
1. vmex fixed-boundary solve -> essos.fields.Vmec
  converged = True, ns = 50, aspect = 6.000
  fixed: LCFS transfer error 1.11e-16 m
  fixed: 4 field lines x 29.7 toroidal turns at s=0.5
  fixed: iota traced 0.419167 +- 1.6e-05  vs wout 0.419155  (rel. 2.9e-05)

2. essos.coils.Coils -> vj.MgridField -> vmex free boundary
  16 filaments tabulated onto 96x32x96 (R, phi, Z)
  converged = True, ns = 50, aspect = 6.000, LCFS solved for by NESTOR
  tabulation vs direct Biot-Savart on the LCFS: 2.72e-05 median, 1.02e-04 max (relative)
  free LCFS sits 3.1 cm from the prescribed one (18% of the minor radius); |B| on axis 0.263 T against 0.882 T

3. the coil-held equilibrium goes back across the same seam
  free: LCFS transfer error 1.11e-16 m
  free: 4 field lines x 8.9 toroidal turns at s=0.5
  free: iota traced 0.418149 +- 3.5e-04  vs wout 0.418149  (rel. 4.5e-07)
```

The LCFS transfer error is a machine-precision identity: ESSOS' `to_xyz`
rebuilding the surface vmex wrote. The iota agreement is the independent check —
ESSOS integrates field lines through the transferred tables, vmex computed
`iotaf` from force balance.

## Local verification

GitHub Actions is out of minutes on this repository, so everything below was run
locally on macOS 14.4, MacPorts Python 3.11, jax/jaxlib 0.9.2, ESSOS 0.17.dev,
`VMEX_COMPILATION_CACHE=disabled`.

| Check | Result |
| --- | --- |
| `ruff check .` | clean |
| `mypy vmex` | no issues in 68 source files |
| `tools/check_docs_prose.py` | prose and media gates clean |
| `sphinx -b html docs` | build succeeded, no warnings; new page renders |
| `pytest tests/test_mgrid.py` | 20 passed, 2 skipped |
| `pytest tests/test_tracing.py` | 6 passed |
| `pytest tests/test_package_api.py tests/test_docs_nav.py tests/test_capability_docs.py` | 28 passed, 3 skipped |
| `pytest tests/test_test_manifest.py` | 5 passed |
| `RUN_FULL=1 pytest tests/test_examples.py::test_vmex_essos_workflow` | passed, 50 s |
| `VMEX_EXAMPLES_CI=1 examples/vmex_essos_workflow.py` | 54--106 s across runs |
| `examples/vmex_essos_workflow.py` (full) | 141--239 s across runs, figure written |

Three new fast nodes are registered in the `pr-physics-field` selector:
`test_from_coils_defaults_bracket_the_coil_set`,
`test_from_coils_batches_the_essos_single_point_biot_savart` and
`test_essos_field_handoff_matches_the_file_route`. The example test itself is
`@pytest.mark.full`, so it stays in `full-examples` where the module already
lives; adding a `full`-marked node to a PR selector would be inert.

Not verified: GPU, and `examples/free_boundary_essos_coils.py` end to end (its
tabulation is byte-identical by construction -- same bounds, same resolution,
same batched values -- and `test_from_coils_batches_the_essos_single_point_biot_savart`
pins that equality against explicit per-point Biot-Savart calls).

### CODE REVIEW

**Comment by rogeriojorge** [comment]
Local CI matrix (Actions out of minutes), branch current with main: quality x6 (ruff, mypy, build, prose, docs, conf.py), fast, all 9 physics selectors, device — TOTALS pass=17 fail=0. Independently spot-checked earlier: 31 passed/2 skipped on the touched test files, and the loop-vs-vmap speedup re-derived at 270x on a stand-in kernel.
